api/python/t4/snapshots.py

```diff
@@ -160,12 +195,22 @@ def hash_file(readable_file):
 
     return hasher.hexdigest()
 
-def dereference_physical_key(physical_key):
-    ty = physical_key['type']
-    if ty == PhysicalKeyType.LOCAL.name:
-        return open(physical_key['path'])
+def read_physical_key(physical_key):
+    # TODO: Stream the data.
+    ty = PhysicalKeyType(physical_key['type'])
+    url = urllib.parse.urlparse(physical_key['path'])
+    if ty == PhysicalKeyType.LOCAL:
+        if url.scheme != 'file':
+            raise ValueError('Unexpected scheme: %r' % url.scheme)
+        with open(url.path, 'rb') as fd:
+            return fd.read()
+    elif ty == PhysicalKeyType.S3:
+        if url.scheme != 's3':
+            raise ValueError('Unexpected scheme: %r' % url.scheme)
+        return download_bytes(url.netloc + url.path)[0]
+    else:
+        assert False
```

Comment:
Raise an exception with unsupport physical key type (maybe "unsupported storage" is more user friendly)?